server/parsec/components/auth.py

```diff
@@ -155,6 +162,77 @@ def verify_signature(self, verify_key: VerifyKey) -> bool:
             return False
 
 
+class AccountAuthenticationType(enum.Enum):
+    Password = b"password"
+
+
+@dataclass
+class AccountAuthenticationToken:
```

Comment:
I think we shouldn't make this structure generic, and instead only focus on the email-based authentication.

Any non email-based authentication method will need a different cooking anyway (since the email field is replaced by something else, the `generate_raw` will take different argument, and the `from_raw` will cook differently)

So in the end it's just better to:
- call this class `AccountPasswordAuthenticationToken`
- change the `HEADER` for something like `PARSEC-PASSWORD-HMAC-BLAKE2B`

This way it will be trivial to add different authentication type and just know which one to use according to the `HEADER` value

Reply:
done